feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/presentation/pools/selectPool/SelectCustomPoolViewModel.kt

```diff
@@ -1,40 +1,123 @@
+    private val setupStakingTypeSelectionMixinFactory: SetupStakingTypeSelectionMixinFactory,
+    private val addressIconGenerator: AddressIconGenerator,
+    private val payload: SelectCustomPoolPayload,
+    private val resourceManager: ResourceManager,
+    private val selectNominationPoolInteractor: SelectingNominationPoolInteractor,
+    private val chainRegistry: ChainRegistry,
+    private val externalActions: ExternalActions.Presentation,
+) : BaseViewModel(), ExternalActions by externalActions {
 
-    val poolModelsFlow = emptyFlow<List<PoolStakeTargetModel>>()
+    private val stakingOption = async(Dispatchers.Default) {
+        val chainWithAsset = chainRegistry.chainWithAsset(payload.chainId, payload.assetId)
+        createStakingOption(chainWithAsset, payload.stakingType)
+    }
+
+    private val setupStakingTypeSelectionMixin = setupStakingTypeSelectionMixinFactory.create(viewModelScope)
+
+    private val poolsFlow = flowOf { selectNominationPoolInteractor.getSortedNominationPools(stakingOption(), viewModelScope) }
+
+    private val nominationPoolRecommenderFlow = flowOf { nominationPoolRecommenderFactory.create(stakingOption(), viewModelScope) }
+
+    private val selectedPoolFlow = MutableStateFlow<NominationPool?>(null)
+
+    val poolModelsFlow = combine(poolsFlow, selectedPoolFlow) { allPools, selectedPool ->
+        convertToModels(allPools, selectedPool)
+    }
         .inBackground()
         .share()
 
-    val selectedTitle = emptyFlow<String>()
+    val selectedTitle = poolsFlow
+        .map { resourceManager.getString(R.string.select_custom_pool_active_pools_count, it.size) }
         .shareInBackground()
 
-    val fillWithRecommendedEnabled = flowOf { false }
+    val fillWithRecommendedEnabled = combine(nominationPoolRecommenderFlow, selectedPoolFlow) { allPools, selectedPool ->
+        allPools.recommendedPool().id != selectedPool?.id
+    }
         .share()
 
-    fun backClicked() {
-        router.back()
+    init {
+        setupStakingTypeSelectionMixin.editableSelectionFlow
+            .onEach {
+                selectedPoolFlow.value = it.selection.asPoolSelection()?.pool
+            }.launchIn(this)
     }
 
-    fun poolInfoClicked(poolStakeModel: PoolStakeTargetModel) = launch {
+    fun backClicked() {
+        router.back()
     }
 
-    fun poolClicked(poolStakeModel: PoolStakeTargetModel) {
+    fun poolClicked(poolItem: PoolRvItem) {
+        launch {
+            val pool = getPoolById(poolItem.id) ?: return@launch
+            setupStakingTypeSelectionMixin.selectNominationPoolAndApply(pool)
+            router.finishSetupPoolFlow()
+        }
     }
 
     fun searchClicked() {
     }
 
     fun selectRecommended() {
+        launch {
+            val recommendedPool = nominationPoolRecommenderFlow.first().recommendedPool()
+            setupStakingTypeSelectionMixin.selectNominationPoolAndApply(recommendedPool)
+            router.finishSetupPoolFlow()
```

Comment:
i beleive we should mark such a selection with `contentRecommended=true`, however current logic marks it with `false` I beleive